distributed/tests/test_zstd.py

```diff
@@ -0,0 +1,52 @@
+def test_zstd():
+    import zstandard, zstandard as zstd, six, cffi, dask, distributed.protocol.compression as compr
+
+    original_data = b"".join(
+        [
+            b"\xe6\x88\x91\xe6\x98\xaf\xe4\xb8\xad\xe5\x9b\xbd\xe4\xba\xba\xe6\xb0",
+            b"\x91\xe7\x9a\x84\xe5\x84\xbf\xe5\xad\x90\xef\xbc\x8c\xe6\x88\x91\xe6",
+            b"\xb7\xb1\xe6\x83\x85\xe5\x9c\xb0\xe7\x83\xad\xe7\x88\xb1\xe7\x9d\x80",
+            b"\xe6\x88\x91\xe7\x9a\x84\xe7\xa5\x96\xe5\x9b\xbd\xe5\x92\x8c\xe4\xba",
+            b"\xba\xe6\xb0\x91\xe3\x80\x82",
+        ]
+    )
+
+    zstd_config_kwargs = [
+        # 'cffi' is NOT support by "pip install zstandard"
+        # but AVAILABLE by manual install zstandard with "python setup.py install"
+        ("distributed.comm.zstd.import_policy", ["default", "cffi_fallback", "cext"]),
+        ("distributed.comm.zstd.level", range(1, 23)),
+        ("distributed.comm.zstd.write_checksum", [False, True]),
+        ("distributed.comm.zstd.write_content_size", [True]),
+        ("distributed.comm.zstd.write_dict_id", [False, True]),
+        ("distributed.comm.zstd.threads", range(-1, 100, 1)),
+    ]
+
+    for cfg, values in zstd_config_kwargs:
```

Comment:
I would normally prefer to use parametrized tests here, but I think that's up to you and @mrocklin if that's needed

Reply:
@mrocklin @crepererum 

To github.com:abaelhe/distributed.git
   00703010..6db2347e  patch-5 -> patch-5
(qianxin_venv) Abael:distributed abaelhe$ pytest  distributed/tests/test_zstd.py  
=============================================================== test session starts ===============================================================
platform darwin -- Python 3.7.4, pytest-5.1.1, py-1.8.0, pluggy-0.12.0 -- /www/workspace/qianxin/qianxin_venv/bin/python3.7
cachedir: .pytest_cache
rootdir: /www/workspace/qianxin/smart_crawler/distributed, inifile: setup.cfg
collected 6 items                                                                                                                                 

distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.import_policy-values0] PASSED                                               [ 16%]
distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.level-values1] PASSED                                                       [ 33%]
distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.write_checksum-values2] PASSED                                              [ 50%]
distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.write_content_size-values3] PASSED                                          [ 66%]
distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.write_dict_id-values4] PASSED                                               [ 83%]
distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.threads-values5] PASSED                                                     [100%]

============================================================ slowest 10 test durations ============================================================
0.22s call     distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.threads-values5]
0.06s call     distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.level-values1]
0.01s call     distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.import_policy-values0]
0.01s call     distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.write_dict_id-values4]
0.01s call     distributed/tests/test_zstd.py::test_zstd[distributed.comm.zstd.write_checksum-values2]

(0.00 durations hidden.  Use -vv to show these durations.)
================================================================ 6 passed in 0.32s ================================================================
(qianxin_venv) Abael:distributed abaelhe$